packages/vitest/src/node/core.ts

```diff
@@ -592,6 +592,9 @@ export class Vitest {
       id = slash(id)
       updateLastChanged(id)
       if (await this.isTargetFile(id)) {
+        const projects = await this.getProjectsByFiles([id])
+        this.projectsTestFiles.set(id, new Set(projects))
```

Comment:
Are you sure this will even be called? As far as I can see it works only if there is a single root workspace, because `this.isTargetFile` checks only the root config.

Reply:
I've been testing this without workspaces only. At least the added test case fails without adding the newly created test file to `this.projectsTestFiles`.